velox/functions/prestosql/aggregates/DecimalAggregate.h

```diff
@@ -231,6 +233,7 @@ class DecimalAggregate : public exec::Aggregate {
         auto decodedIndex = decodedPartial_.index(0);
         auto serializedAccumulator =
             intermediateFlatVector->valueAt(decodedIndex);
+        clearNull(group);
```

Comment:
Maybe wrap it in `if (rows.hasSelections())`

Reply:
I would prefer to keep it as is or remove it completely.

Reply:
I would have to add a test once I add `if (rows.hasSelections())` and I don't know how to trigger this particular case.

Reply:
It's masked aggregation.  Try filter out all the rows.

Reply:
I don't know what masked aggregation is.
I updated the code with your suggestion but I don't know how to test `addSingleGroupIntermediateResults` method. I checked and none of the existing tests verify that particular logic.

Please note that I don't feel confident making the additional changes suggested in the review without having a test to verify them. If you think the change is good, then we can merge the PR. If more tests are needed, I would either prefer to revert to the original change or add more tests. In the latter case, please advise on how I can test those particular sections in the code. Thanks.

Reply:
I meant how to trigger `decodedRaw_.isConstantMapping()` in `addSingleGroupRawInput`. I figured how I can do it on my own: you need constant expressions in the aggregate functions, e.g. `avg(10.2)`. However, it not possible to do with PlanBuilder because all the input into intermediate or final aggregation will not be raw anymore and will not have constant mapping - all my trials were unsuccessful. For example, I can trigger `decodedRaw_.isConstantMapping()` in `addSingleGroupRawInput` method easily but not for intermediate results.

At this point, I have decided to ignore the test for this particular scenario.

Reply:
Yeah then just skip test this case for now